datafusion/physical-expr/src/aggregate/approx_distinct.rs

```diff
@@ -219,7 +219,11 @@ macro_rules! default_accumulator_impl {
     () => {
         fn merge_batch(&mut self, states: &[ArrayRef]) -> Result<()> {
             assert_eq!(1, states.len(), "expect only 1 element in the states");
-            let binary_array = states[0].as_any().downcast_ref::<BinaryArray>().unwrap();
+            let binary_array = states[0].as_any().downcast_ref::<BinaryArray>().ok_or(
```

Comment:
It might be useful to introduce a helper function or macro for downcasting arrays to expect tyes. I think there might be one in arrow somewhere that I added a long time ago. I will try and find it.

Reply:
There is already a macro defined in this file that looks like what we need

```
macro_rules! downcast_value {
    ($Value: expr, $Type: ident, $T: tt) => {{
        $Value[0]
            .as_any()
            .downcast_ref::<$Type<T>>()
            .ok_or_else(|| {
                DataFusionError::Internal(format!(
                    "could not cast value to {}",
                    type_name::<$Type<T>>()
                ))
            })?
    }};
}
```

Reply:
@andygrove Fixed! Actually I expanded the macro and put it in `datafusion-common` since it is everywhere.